dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -469,49 +469,51 @@ subroutine dglc_datamode_noevolve_restart_write(model_meshes, case_name, &
+    ! write restart info to rpointer file
+    if (my_task == main_task) then
+       open(newunit=nu, file=trim(rpfile)//trim(inst_suffix), form='formatted')
+       write(nu,'(a)') rest_file_model
+       close(nu)
+       write(logunit,'(a,2x,i0,2x,i0)')' writing with no streams '//trim(rest_file_model), ymd, tod
+    endif
+
+    ! write data model restart data
+    rcode = pio_createfile(pio_subsystem, pioid, io_type, trim(rest_file_model), pio_clobber)
+    
     do ns = 1,num_icesheets
-       ! Determine gindex for this ice sheet
-       call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-       allocate(gindex(lsize))
-       call ESMF_DistGridGet(distGrid, localDe=0, seqIndexList=gindex, rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
-
        ! Need to explicitly write restart since noevolve mode does not read a stream
        write(cnum,'(i0)') ns
 
-       call shr_cal_datetod2string(date_str, ymd, tod)
-       write(rest_file_model ,"(7a)") trim(case_name),'.','dglc',trim(inst_suffix),'.r.',trim(date_str),'.nc'
-
-       ! write restart info to rpointer file
-       if (my_task == main_task) then
-          open(newunit=nu, file=trim(rpfile)//trim(inst_suffix), form='formatted')
-          write(nu,'(a)') rest_file_model
-          close(nu)
-          write(logunit,'(a,2x,i0,2x,i0)')' writing with no streams '//trim(rest_file_model), ymd, tod
-       endif
-
-       ! write data model restart data
-       rcode = pio_createfile(pio_subsystem, pioid, io_type, trim(rest_file_model), pio_clobber)
        rcode = pio_def_dim(pioid, '_nx'//trim(cnum), nx_global(ns), dimid2(1))
        rcode = pio_def_dim(pioid, '_ny'//trim(cnum), ny_global(ns), dimid2(2))
        rcode = pio_def_var(pioid, 'flgl_rofi'//cnum, PIO_DOUBLE, (/dimid2/), varid)
        rcode = pio_put_att(pioid, varid, "_FillValue", shr_const_spval)
        rcode = pio_set_fill(pioid, PIO_FILL, oldmode)
        rcode = pio_put_att(pioid, pio_global, "version", "nuopc_data_models_v0")
-       rcode = pio_enddef(pioid)
+    enddo
+    rcode = pio_enddef(pioid)
 
+    do ns = 1,num_icesheets
+
+       ! Determine gindex for this ice sheet
+       call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       allocate(gindex(lsize))
+       call ESMF_DistGridGet(distGrid, localDe=0, seqIndexList=gindex, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       
        call pio_initdecomp(pio_subsystem, pio_double, (/nx_global(ns),ny_global(ns)/), gindex, pio_iodesc)
        call pio_write_darray(pioid, varid, pio_iodesc, Fgrg_rofi(ns)%ptr, rcode, fillval=shr_const_spval)
```

Comment:
This is incorrect in that varid should be an array and not a scalar. How does this ever work for multiple ice sheets?

Reply:
This refers to line 510 above which was not modified and I think should be.